### PR TITLE
Fix immediate apply mode for maxResults text field in filter

### PR DESCRIPTION
**Pull request by daring2**
Fix the problem described in issue #2572

### CODE REVIEW

File: modules/gui/src/com/haulmont/cuba/gui/components/filter/FilterDelegateImpl.java [hunk 1]
```diff
@@ -496,25 +496,28 @@ protected void createMaxResultsLayout() {
         maxResultsTextField.setStyleName("c-maxresults-input");
         maxResultsTextField.setMaxLength(4);
         maxResultsTextField.setWidth(theme.get("cuba.gui.Filter.maxResults.width"));
+        maxResultsTextField.addValueChangeListener(this::onMaxResultsChange);
 
         maxResultsLookupField = maxResultsFieldHelper.createMaxResultsLookupField();
         maxResultsLookupField.setStyleName("c-maxresults-select");
+        ((LookupField<Integer>) maxResultsLookupField).addValueChangeListener(this::onMaxResultsChange);
```
Comment (Flaurite) [inline modules/gui/src/com/haulmont/cuba/gui/components/filter/FilterDelegateImpl.java]: Lets add `Integer` type to `maxResultsLookupField` and parameterized:
- `MaxResultsFieldHelper#createMaxResultsLookupField()`
- `MaxResultsFieldHelper#setUpMaxResultsLookupField()` methods too. 

Thus there will no need to cast to `LookupField<Integer>`

Comment (daring2) [reply modules/gui/src/com/haulmont/cuba/gui/components/filter/FilterDelegateImpl.java]: Fixed

File: modules/gui/src/com/haulmont/cuba/gui/components/LookupField.java [hunk 2]
```diff
@@ -33,6 +33,7 @@
     String NAME = "lookupField";
 
     TypeToken<LookupField<String>> TYPE_STRING = new TypeToken<LookupField<String>>(){};
+    TypeToken<LookupField<Integer>> TYPE_INTEGER = new TypeToken<LookupField<Integer>>(){};
```
Comment (Flaurite) [inline modules/gui/src/com/haulmont/cuba/gui/components/LookupField.java]: Thank you for your intention, but in bugfix issues we do not provide additional abilities :)
Remove it and just use `LookupField#of(Class<T> valueClass)` like this:
```
uiComponents.create(LookupField.of(Integer.TYPE))
```

In general its pretty good, after fixes we will merge it.

Comment (daring2) [reply modules/gui/src/com/haulmont/cuba/gui/components/LookupField.java]: ok, fixed

**Pull request merged**